include/mantella_bits/optimisationAlgorithm/covarianceMatrixAdaptationEvolutionStrategy.hpp

```diff
@@ -0,0 +1,108 @@
+#pragma once
+
+// Armadillo
+#include <armadillo>
+
+// Mantella
+#include "mantella_bits/optimisationAlgorithm.hpp"
+
+namespace mant {
+  class OptimisationProblem;
+}
+
+namespace mant {
+  class CovarianceMatrixAdaptationEvolutionStrategy : public OptimisationAlgorithm {
+   public:
+    explicit CovarianceMatrixAdaptationEvolutionStrategy();
+
+    void initialise(
+        const arma::uword numberOfDimensions,
+        const arma::Mat<double>& initialParameters) override;
+
+    //using OptimisationAlgorithm::optimise;
+
+    void optimise(
+        OptimisationProblem& optimisationProblem);
+    void optimise(
+        OptimisationProblem& optimisationProblem,
+        const arma::uword popSize);
+    void optimise(
+        OptimisationProblem& optimisationProblem,
+        const arma::Col<double>& xMean);
+    void optimise(
+        OptimisationProblem& optimisationProblem,
+        const arma::Mat<double>& initialParameters);
+    void optimise(
+        OptimisationProblem& optimisationProblem,
+        const arma::Col<double>& xMean,
+        const arma::uword popSize);
+
+    void setInitialStepSize(const double sigma); //insigma
+    double getInitialStepSize();
+    double getStepSize();
+    void setPopulationSize(const arma::uword popSize, const arma::uword numberOfDimensions);
+    arma::uword getPopulationSize();
+    void setActiveCMA(bool activeCma); //activeCMA
```

Comment:
Die Nutzung von activeCMA war besser, nicht? In diesem Fall würde ich dies als Standardeinstellung vorhsehen und die Option dafür entfernen.